vector/src/main/java/im/vector/app/features/displayname/Extension.kt

```diff
@@ -24,8 +24,26 @@ fun MatrixItem.getBestName(): String {
         // Best name is the id, and we keep the displayName of the room for the case we need the first letter
         id
     } else {
-        displayName
+//        displayName
+//                ?.takeIf { it.isNotBlank() }?.replace("[TG] ","")?.replace("$","")
+//                ?: VectorMatrixItemDisplayNameFallbackProvider.getDefaultName(this).replace("[TG] ","").replace("$","")
+        val modifiedDisplayName = displayName
                 ?.takeIf { it.isNotBlank() }
-                ?: VectorMatrixItemDisplayNameFallbackProvider.getDefaultName(this)
+                ?.let {
+                    when {
+                        it.startsWith("[TG] ") -> {
+                            it.replaceFirst("[TG] ", "") // Remove [TG] from the beginning
+                        }
+                        it.contains("$") -> {
+                            it.replace("$", "") // Remove $ from the string
```

Comment:
I guess all these replacements should only change the first occurance as the name may contain multiple '$$$'

Reply:
I think that block change is redundant anyway, there is another "replacement" again on the resulting string below